homeassistant/components/sensor/turn_touch.py

```diff
@@ -0,0 +1,71 @@
+"""
+Provides a sensor to track the battery level of a Turn Touch remote.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.turn_touch/
+"""
+
+import logging
+
+from homeassistant.const import DEVICE_CLASS_BATTERY
+from homeassistant.components import turn_touch
+from homeassistant.components.sensor import ENTITY_ID_FORMAT
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = [turn_touch.DOMAIN]
+
+SENSOR_NAME = '{remote_name} Battery'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Turn Touch remote battery sensor platform."""
+    devices = []
+    for turntouch_device in hass.data[turn_touch.DATA_KEY]['devices'].values():
+        devices.append(TurnTouchBatterySensor(turntouch_device))
+    add_devices(devices)
+
+
+class TurnTouchBatterySensor(Entity):
+    """Representation of a Turn Touch remote's battery."""
+
+    def __init__(self, turntouch_device):
+        """
+        Initialize the Turn Touch battery level sensor.
+
+        This expects that the remote has already been initialized by the
+        turn_touch component.
+        """
+        _LOGGER.debug('Initializing turn touch battery sensor.')
+        self.turn_touch = turntouch_device
+        self._battery_level = self.turn_touch.get_battery()
```

Comment:
move this into `def update(self)` and call add_devices(..., True)

Reply:
@pvizeli Just one question about your comment:

> Remove this reconnect logic from functions

Those functions aren't called from `@property` methods (`name()`, `state()`), only from `TurnTouchRemote__init_.()` and `TurnTouchBatterySensor.update()`. My understanding is that Home Assistant initializes components and calls `update()` from a separate thread already, so wrapping it in another thread seems redundant. Is that wrong?